GlobalQuakeServer/src/main/java/gqserver/websocketserver/Clients.java

```diff
@@ -0,0 +1,149 @@
+package gqserver.websocketserver;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+
+import org.tinylog.Logger;
+
+import globalquake.core.Settings;
+
+
+
+public class Clients {
+    private ScheduledExecutorService pingExecutor;
+
+    // IP:PORT -> Client
+    private Hashtable<String, Client> clients;
+    
+    //IP -> Integer
+    private Hashtable<String, Integer> uniqueIPConnectionCounts;
+    
+    private static Clients instance = new Clients();
+    private Clients() {
+        pingExecutor = Executors.newScheduledThreadPool(4);
+        clients = new Hashtable<String, Client>();
+        uniqueIPConnectionCounts = new Hashtable<String, Integer>();
+    }
+
+    public int getCountForIP(String ip) {
+        int count = 0;
+        try{
+            count = uniqueIPConnectionCounts.get(ip);
+        }
+        catch(Exception e) {}
+        
+        return count;
+    }
+
+    /**
+     * Increment the connection count for the given unique IP address
+     * @param address
+     * @return The new connection count for the given IP address
+     */
+    private int incrementConnectionCount(String address) {
+        int count = 0;
+
+        synchronized (uniqueIPConnectionCounts) {

```

Comment:
the synchronizations here seem a bit sketchy to me :skull: I would be quite worried about deadlocks or race conditions from my experience. Perhaps you can use ConcurrentHashMap to do the job?